Failure was:

```
[2021-09-06T19ː26ː29.273Z] FAILURE: Build failed with an exception. [2021-09-06T19ː26ː29.273Z] [2021-09-06T19ː26ː29.273Z] * What went wrong: [2021-09-06T19ː26ː29.273Z] Execution failed for task 'ːstreamsːstreams-scala:compileScala'. [2021-09-06T19ː26ː29.273Z] > Timeout waiting to lock zinc-1.3.5_2.12.14_8 compiler cache (/home/jenkins/.gradle/caches/7.2/zinc-1.3.5_2.12.14_8). It is currently in use by another Gradle instance. [2021-09-06T19ː26ː29.273Z] Owner PID: 6057 [2021-09-06T19ː26ː29.273Z] Our PID: 6101 [2021-09-06T19ː26ː29.273Z] Owner Operation: [2021-09-06T19ː26ː29.273Z] Our operation: [2021-09-06T19ː26ː29.273Z] Lock file: /home/jenkins/.gradle/caches/7.2/zinc-1.3.5_2.12.14_8/zinc-1.3.5_2.12.14_8.lock [2021-09-06T19ː26ː29.273Z] 
```

--- a/clients/src/main/java/org/apache/kafka/common/network/Selector.java
+++ b/clients/src/main/java/org/apache/kafka/common/network/Selector.java
@@ -1275,7 +1275,7 @@ public class Selector implements Selectable, AutoCloseable {
         }
 
         /**
-         * This method generates `time-total` metrics with a couple of errors, no `-ns` suffix and no dash between basename
+         * This method generates `time-total` metrics but has a couple of deficiencies: no `-ns` suffix and no dash between basename
          * and `time-toal` suffix.
          * @deprecated use {{@link #createIOThreadRatioMeter(Metrics, String, Map, String, String)}} for new metrics instead
          */

--- a/docs/ops.html
+++ b/docs/ops.html
@@ -1892,12 +1892,12 @@ $ bin/kafka-acls.sh \
       </tr>
       <tr>
         <td>io-wait-time-ns-total</td>
-        <td>The total time the I/O thread spent waiting in nanoseconds</td>
+        <td>The total time the I/O thread spent waiting in nanoseconds.</td>
         <td>kafka.[producer|consumer|connect]:type=[producer|consumer|connect]-metrics,client-id=([-.\w]+)</td>
       </tr>
       <tr>
         <td>io-waittime-total</td>
-        <td><b>*Deprecated*</b> The total time the I/O thread spent waiting in nanoseconds</td>
+        <td><b>*Deprecated*</b> The total time the I/O thread spent waiting in nanoseconds. Replacement is <code>io-wait-time-ns-total</code>.</td>
         <td>kafka.[producer|consumer|connect]:type=[producer|consumer|connect]-metrics,client-id=([-.\w]+)</td>
       </tr>
       <tr>
@@ -1912,12 +1912,12 @@ $ bin/kafka-acls.sh \
       </tr>
       <tr>
         <td>io-time-ns-total</td>
-        <td>The total time the I/O thread spent doing I/O in nanoseconds</td>
+        <td>The total time the I/O thread spent doing I/O in nanoseconds.</td>
         <td>kafka.[producer|consumer|connect]:type=[producer|consumer|connect]-metrics,client-id=([-.\w]+)</td>
       </tr>
       <tr>
         <td>iotime-total</td>
-        <td><b>*Deprecated*</b> The total time the I/O thread spent doing I/O in nanoseconds</td>
+        <td><b>*Deprecated*</b> The total time the I/O thread spent doing I/O in nanoseconds. Replacement is <code>io-time-ns-total</code>.</td>
         <td>kafka.[producer|consumer|connect]:type=[producer|consumer|connect]-metrics,client-id=([-.\w]+)</td>
       </tr>
       <tr>
@@ -2094,7 +2094,7 @@ $ bin/kafka-acls.sh \
       </tr>
       <tr>
         <td>bufferpool-wait-time-total</td>
-        <td><b>*Deprecated*</b> The total time an appender waits for space allocation in nanoseconds.</td>
+        <td><b>*Deprecated*</b> The total time an appender waits for space allocation in nanoseconds. Replacement is <code>bufferpool-wait-time-ns-total</code></td>
         <td>kafka.producer:type=producer-metrics,client-id=([-.\w]+)</td>
       </tr>
       <tr>

--- a/docs/upgrade.html
+++ b/docs/upgrade.html
@@ -24,7 +24,8 @@
     <li>Apache Kafka supports Java 17.</li>
     <li>The following metrics have been deprecated: <code>bufferpool-wait-time-total</code>, <code>io-waittime-total</code>,
         and <code>iotime-total</code>. Please use <code>bufferpool-wait-time-ns-total</code>, <code>io-wait-time-ns-total</code>,
-        and <code>io-time-ns-total</code> instead.</li>
+        and <code>io-time-ns-total</code> instead. See <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-773%3A+Differentiate+consistently+metric+latency+measured+in+millis+and+nanos">KIP-773</a>
+        for more details.</li>
 </ul>
 
 <h5><a id="upgrade_300_notable" href="#upgrade_300_notable">Notable changes in 3.0.0</a></h5>

